Fix problem when call interact with escape_character=None #132

marcosps opened the pull request:
Hi @jquast ,

thanks for the tip about virtualenv. I coudl test the change, and it seems it worked as expected. If you find something that can be better, please tell me.

jquast commented:
Thank you for your contribution Marcos! 

I would like the following changes:
- explicitly check if ``escape_character is None``
- docstring describing that "escaping may be disabled all-together by setting ``escape_character=None``.
- use long-hand ``if`` statement instead of shorthand.

More difficult, I would like to see a test -- the file ``tests/test_interact.py`` provides one for called ``test_interact_escape``, the supporting program ``interact.py`` should be provided to evaluate whether an optional argument ``--noescape`` is used, which when it is, instead uses ``p.interact(escape_character=None)``. And a supporting test that executes this script with that argument and asserts that the default escape character is not evaluated.

I trust the code works fine, the purpose of such tests is to ensure changes or refactoring in the future does not accidentally break your feature.

You can do any or none of these, I will complete them otherwise. Thanks again!

marcosps commented:
Hi @jquast ,

thanks for your comments. I'll rework my patch and send it again ASAP.

marcosps commented:
And I'll work in all things that you pointed. Let all this to me :)

marcosps commented:
Ops, now it seems the unittest is getting an error. I'll try to work on those today @jquast .

BTW, I addressed your comments about pty_spawn.py, maybe some adjustments in test_interact should fix this...

jquast commented on pexpect/pty_spawn.py:
```diff
@@ -677,11 +677,12 @@ def interact(self, escape_character=chr(29),
-        the ESC character. ASCII 29 was chosen for historical merit because
-        this is the character used by 'telnet' as the escape character. The
-        escape_character will not be sent to the child process.
+        escape_character this method will stop. Escaping may be disableall-together
```
I realize my quick draft resulted in this, but please: ``"disableall-together" => "disabled entirely"``

jquast commented:
It looks like you just got lucky about the failing test -- it was an intermittent failure, I re-ran it and it was fine.

(As travis has lost the history, it was in the interact spawn test -- expecting an EOF but not receiving one.  The last buffer however showed something like a final "<in> end of file\r\nno coverage will be collected." sort of text that indicates that it did, in fact, exit cleanly.

Also, noticed github was not sending push notifications to teamcity for pull requests, manually added a polling VCS trigger for pull requests. It is building the SunOS/OSX/Linux matrix now.)

marcosps commented:
Hi @jquast ,

sorry about the comment. It was a blind change. Fixed now. So, you think the change is ready to be pushed? I can send the changes in tests in another PRs.

jquast commented:
sorry for not replying sooner, but it absolutely requires tests, i meant to write them but i've been hung up
